syntax = "proto2";

package rv.protos;

import "google/protobuf/struct.proto";

message VectorSourceConfig {
    message MBTiles {
        required string uri = 1;
    }

    message GeoJSON {
        required string uri = 1;
    }

    oneof vector_source_config {
        MBTiles mbtiles = 1;
        GeoJSON geojson = 2;
        google.protobuf.Struct custom_config = 3;
    }

    optional google.protobuf.Struct class_id_to_filter = 4;
    required string source_type = 5;
}
